source/nanoFramework.Hardware.Esp32/TouchPad/TouchPadBase.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections;
+using System.Runtime.CompilerServices;
+
+namespace nanoFramework.Hardware.Esp32.TouchPad
+{
+  /// <summary>
+  /// Base class for toch pad sensors: interrupt or polling driven, using single or multiple gpio pins
+  /// </summary>
+  public abstract class TouchPadBase // : IDisposable
+  {
+    //code review: Gpio​Pin uses it, but it's internal to Windows.Devices.Gpio  - please review if it applicable here
+    //private static GpioPinEventListener s_eventListener;
+
+    // this is used as the lock object 
+    // a lock is required because multiple threads can access the GpioPin
+    private readonly object _syncLock = new object ();
+
+    protected readonly int _pinNumber;
+		protected readonly int _touchPadIndex;
+
+		/// <summary>
+		/// Map of GPIO to touch pad number. 
+		/// ESP32 offers up to 10 capacitive IOs that detect changes in capacitance on touch sensors due to finger contact or proximity. 
+		/// </summary>
+		private static readonly Hashtable GpioTouchPadMap = new Hashtable()
+    {
+      {4, 0}, {0, 1}, {2,  2}, {27, 7}, {15, 3}, {13, 4}, {12, 5}, {14, 6}, {33, 8}, {32, 9}
+    };
+
+
+    protected TouchPadBase(int pinNumber, TouchPadBaseConfig config)
+    {
+			if (!GpioTouchPadMap.Contains(_pinNumber))
+				throw new Exception($"Pin {_pinNumber} is not a valid touch pad pin.");
+
+			if(config == null)
+				throw new ArgumentNullException(nameof(config));
```

Comment:
this doesn't work in nF

Reply:
I was pointing out that `nameof(pinNumber)`  isn't supported in nF. You have to drop it in the call to ArgumentException constructor.